ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/platform.py

```diff
@@ -162,23 +162,46 @@ def default_interfaces(self: SharedUtils) -> dict:
         default_interfaces set based on default_interfaces
         """
         default_interfaces = get(self.hostvars, "default_interfaces", default=[])
-
         device_platform = default(self.platform, "default")
 
+        # Re-order default_interfaces:
+        # #  platform  type
+        # 1  abc       xyz
+        # 2  abc       default
+        # 3  default   xyz
+        # 4  default   default
+        default_interfaces.sort(key=self.sort_default_interfaces)
+
         # First look for a matching default interface set that matches our platform and type
         for default_interface in default_interfaces:
             for platform in default_interface.get("platforms", []):
-                if search(f"^{platform}$", device_platform) and self.type in default_interface.get("types", []):
+                if search(f"^{platform}$", device_platform) and (
+                    self.type in default_interface.get("types", []) or "default" in default_interface.get("types", [])
+                ):
                     return default_interface
 
         # If not found, then look for a default default_interface that matches our type
         for default_interface in default_interfaces:
             for platform in default_interface.get("platforms", []):
-                if search(f"^{platform}$", "default") and self.type in default_interface.get("types", []):
+                if search(f"^{platform}$", "default") and (self.type in default_interface.get("types", []) or "default" in default_interface.get("types", [])):
```

Comment:
```
search(f"^{platform}$", "default")
```
could be
```
platform == "default"
```

Reply:
Unnecessary since we don't need that 2nd loop.